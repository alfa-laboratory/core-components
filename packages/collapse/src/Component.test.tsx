import React from 'react';
import { render } from '@testing-library/react';

import { Collapse } from './index';

const paragraph = (
    <p style={{ margin: '0 0 16px 0' }}>
        Альфа-Банк, основанный в 1990 году, является универсальным банком, осуществляющим все
        основные виды банковских операций, представленных на рынке финансовых услуг, включая
        обслуживание частных и корпоративных клиентов, инвестиционный банковский бизнес, торговое
        финансирование и т.д.
    </p>
);

// TODO: more tests
describe('Collapse', () => {
    describe('Display tests', () => {
        it('should display with children like boolean or string or others react children type correctly', () => {
            const { container } = render(<Collapse>{paragraph}</Collapse>);

            expect(container).toMatchSnapshot();
        });

        it('should display radio group with one child correctly', () => {
            const { container } = render(
                <Collapse collapsedLabel='Показать'>{paragraph}</Collapse>,
            );

            expect(container).toMatchSnapshot();
        });
    });
});
