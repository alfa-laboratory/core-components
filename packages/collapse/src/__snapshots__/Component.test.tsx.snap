// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse Display tests should display radio group with one child correctly 1`] = `
<div>
  <div
    class="collapse"
  >
    <div
      class="content collapsed"
    >
      <div
        class="contentCase"
      >
        <p
          style="margin: 0px 0px 16px 0px;"
        >
          Альфа-Банк, основанный в 1990 году, является универсальным банком, осуществляющим все основные виды банковских операций, представленных на рынке финансовых услуг, включая обслуживание частных и корпоративных клиентов, инвестиционный банковский бизнес, торговое финансирование и т.д.
        </p>
      </div>
    </div>
    <a
      class="component primary pseudo withAddons label"
    >
      <span>
        <span
          class="text"
        >
          Показать
        </span>
      </span>
      <span
        class="addons"
      >
        <svg
          focusable="false"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M0 24h24V0H0z"
            />
            <path
              d="M16.074 9.646l.708.708-5.068 5.068-5.068-5.068.708-.708 4.36 4.362z"
              fill="#000"
              fill-rule="nonzero"
            />
          </g>
        </svg>
      </span>
    </a>
  </div>
</div>
`;

exports[`Collapse Display tests should display with children like boolean or string or others react children type correctly 1`] = `
<div>
  <div
    class="collapse"
  >
    <div
      class="content collapsed"
    >
      <div
        class="contentCase"
      >
        <p
          style="margin: 0px 0px 16px 0px;"
        >
          Альфа-Банк, основанный в 1990 году, является универсальным банком, осуществляющим все основные виды банковских операций, представленных на рынке финансовых услуг, включая обслуживание частных и корпоративных клиентов, инвестиционный банковский бизнес, торговое финансирование и т.д.
        </p>
      </div>
    </div>
  </div>
</div>
`;
